tests/test_views.py

```diff
@@ -0,0 +1,127 @@
+        """Testing main page for render needed posts."""
+        time_zone = timezone.get_current_timezone()
+        future_post = Post.objects.create(
+            author=self.user,
+            title='future_test',
+            text='superText',
+            created_date=datetime(
+                day=1, month=4,
+                year=3020,
+                tzinfo=time_zone,
+            ),
+            published_date=datetime(
+                day=1,
+                month=4,
+                year=3020,
+                tzinfo=time_zone,
+            ),
+        )
+        post = Post.objects.create(
+            author=self.user, title='Test',
+            text='superText',
+            created_date=datetime(
+                day=1, month=3,
+                year=2020, tzinfo=time_zone,
+            ),
+            published_date=datetime(
+                day=1, month=3,
+                year=2020,
+                tzinfo=time_zone,
+            ),
+        )
+        past_post = Post.objects.create(
+            author=self.user, title='past_est',
+            text='superText',
+            created_date=datetime(
+                day=1, month=4,
+                year=2019,
+                tzinfo=time_zone,
+            ),
+            published_date=datetime(
+                day=1, month=4,
+                year=2019,
+                tzinfo=time_zone,
+            ),
+        )
+        with patch(
+                'django.utils.timezone.now', lambda: datetime(
+                    day=1,
+                    month=1,
+                    year=2020,
+                    tzinfo=time_zone,
+                ),
+        ):
+            response = self.client.get(reverse('post_list'))
+            self.assertListEqual(list(response.context['posts']), [past_post])
+            self.assertNotContains(response, post)
+            self.assertNotContains(response, future_post)
+        with patch(
+                'django.utils.timezone.now', lambda: datetime(
+                    day=1,
+                    month=10,
+                    year=2020,
+                    tzinfo=time_zone,
+                ),
+        ):
+            response = self.client.get(reverse('post_list'))
+            self.assertListEqual(
+                list(response.context['posts']), [
+                    post,
+                    past_post,
+                ],
+            )
+            self.assertNotContains(response, future_post)
+        with patch(
+                'django.utils.timezone.now', lambda: datetime(
+                    day=1,
+                    month=4,
+                    year=3020,
+                    tzinfo=time_zone,
+                ),
+        ):
+            response = self.client.get(reverse('post_list'))
+            self.assertListEqual(
+                list(response.context['posts']), [
+                    future_post,
+                    post,
+                    past_post,
+                ],
+            )
```

Comment:
These case are semantically separate subtests. Could you please try using [`TestCase.subTest()`](
https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests
) for them? Just have a loop over different dates with a list comparison check inside.